File: Python/SubtreeOfAnotherTree.py
class TreeNode:
    def __init__(self,val=0,left=None,right=None) -> None:
        self.val = val
        self.left = left
        self.right = right


    def isSubtree(self,root,subRoot):
        if not subRoot:
            return True
        
        if not root:
            return False
        
        if self.sameTree(root,subRoot):
            return True
        
        return (self.isSubtree(root.left,subRoot) or self.isSubtree(root.right,subRoot))
    
    def sameTree(self,t,s):
        if not t and not s:
            return True
        
        if s and t and s.val == t.val:
            return (self.sameTree(t.left,s.left) and self.sameTree(t.right,s.right))
        
        return False

File: Javascript/SubtreeOfAnotherTree.js
function TreeNode(val,left,right){
    this.val = (val === undefined) ? 0 : val
    this.left = (left === undefined) ? left : val
    this.right = (right === undefined) ? right : val
}


var isSubTree = function(root,subRoot){
    if(subRoot === null){
        return true
    }

    if(root === null){
        return false
    }

    if(sameTree(root,subRoot)){
        return true
    }

    return (isSubTree(root.left,subRoot) || isSubTree(root.right,subRoot))

    function sameTree(t,s){
        if(s=== null && t ===null){
            return true
        }

        if(s !== null && t!==null && s.val === t.val){
            return (sameTree(t.left,s.left) && sameTree(t.right,s.right))
        }

        return false
    }

}